app/__tests__/bcsc-theme/utils/read-file.test.ts

```diff
@@ -0,0 +1,232 @@
+import readFileInChunks from '@/utils/read-file'
+import { Buffer } from 'buffer'
+import RNFS from 'react-native-fs'
+
+jest.mock('react-native-fs', () => ({
+  stat: jest.fn(),
+  read: jest.fn(),
+}))
+
+describe('readFileInChunks', () => {
+  const mockLogger = {
+    debug: jest.fn(),
+    warn: jest.fn(),
+    error: jest.fn(),
+  }
+
+  const mockFilePath = '/some/test/file.mp4'
+
+  beforeEach(() => {
+    jest.clearAllMocks()
+  })
+
+  describe('successful file reading', () => {
+    it('should read a small file in a single chunk', async () => {
+      const testData = 'test file content'
+      const testBuffer = Buffer.from(testData, 'utf8')
+      const base64Data = testBuffer.toString('base64')
+
+      ;(RNFS.stat as jest.Mock).mockResolvedValue({
```

Comment:
fyi you can create a fully typed mocked module using jest.mocked. No changes needed just useful information.

```ts
const RNFSMock = jest.mocked(RNFS)

RNFSMock.exists.mockRsolvedValue(true)
```